### PR TITLE
WordPress.com VIP requested security fixes

### PR DESCRIPTION
**Pull request by moraleida**
## Summary

Addresses issue #901 

## Relevant technical choices

Simple sanitization and validation fixes requested by VIP when deploying to one of their clients.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/Admin/Notice.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ public function render() {
 
 		?>
 		<div id="<?php echo esc_attr( 'googlesitekit-notice-' . $this->slug ); ?>" class="<?php echo esc_attr( $class ); ?>">
-			<?php echo $content; /* phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped */ ?>
+			<?php echo wp_kses_post( $content ); /* phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped */ ?>
```
Comment (felixarntz) [inline includes/Core/Admin/Notice.php]: Do we still need the ignore comment here?

Comment (moraleida) [reply includes/Core/Admin/Notice.php]: Good catch, thanks. Removed it.

**Comment by moraleida** [comment]
> Thanks for the PR @moraleida. While making these changes, we should also ensure that we actually include the `WordPressVIPMinimum` ruleset in our project setup (see #901). Would you mind amending this PR by adding the necessary Composer packages / PHPCS references?

@felixarntz I've added the packages and rules to phpcs per the definition. VIP has 2 different standards, the legacy `WordPressVIPMinimum` and `WordPress-VIP-Go`, which contains the former and a few more sniffs. I've added both and referenced `WordPress-VIP-Go` in `phpcs.xml` so we have a complete check.

File: includes/Core/Authentication/Verification_Tag.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+/**
+ * Class Google\Site_Kit\Core\Authentication\Verification_Tag
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Authentication;
```
Comment (felixarntz) [inline includes/Core/Authentication/Verification_Tag.php]: This file was removed in the meantime, can you re-remove it?

Comment (moraleida) [reply includes/Core/Authentication/Verification_Tag.php]: Removed. I've added you as a collaborator to my repo @felixarntz . Let me know if you need anything else, please. Thank you!

**Comment by tofumatt** [comment]
Looks like these changes introduced a pretty serious bug in #964, so we may want to think about introducing E2E tests that verify the output of the plugin activation notice.

**Comment by felixarntz** [comment]
@tofumatt I added PHPUnit tests in #966 that should cover for it. The problem was that there was no testing of this overall so far.

**Comment by moraleida** [comment]
@felixarntz @tofumatt making a note here that the solution in #966 is likely to prevent this from being used in VIP projects. `$content` is being retrieved from a a dynamically named function in '$this->args['content']` and the fact that kses is being removed to allow it to contain script tags is really likely to strike this down. I believe the solution here is to load the scripts separately.

**Comment by felixarntz** [comment]
@moraleida The escaping / usage of KSES should happen in this dynamic function. Since this is not user-generated content, there is not much purpose KSESing it before output, unless a developer elsewhere messed it up.

We can potentially revisit in the future adding the script through a different means, but for now this is what needs to happen for it to work, and I don't think it's insecure.

**Pull request merged**